docs/02-app/01-building-your-application/06-optimizing/05-scripts.mdx

```diff
@@ -147,8 +147,85 @@ npm run dev
 
 You'll see instructions like these: Please install Partytown by running `npm install @builder.io/partytown`
 
+<AppOnly>
+
+To use the `worker` strategy with the App Router, you need to import the `Partytown` component in your root layout. This ensures that Partytown is properly set up in your application.
+
+```tsx filename="app/layout.tsx" switcher
+import { Partytown } from '@builder.io/partytown/react'
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <html lang="en">
+      <head>
+        <Partytown lib={'/_next/static/~partytown/'} />
+      </head>
+      <body>
+        {/* Layout UI */}
+        <main>{children}</main>
+      </body>
+    </html>
+  )
+}
+```
+
+```jsx filename="app/layout.js" switcher
+import { Partytown } from '@builder.io/partytown/react'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <head>
+        <Partytown lib={'/_next/static/~partytown/'} />
```

Comment:
I think we should avoid users having to add this manually. 

We can handle it similar to how we do for `_document` here https://github.com/vercel/next.js/blob/canary/packages/next/src/pages/_document.tsx#L253

Which could potentially go in `app-router` here https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/app-router.tsx#L603 if partytown support is enabled.

Reply:
> I think we should avoid users having to add this manually.

I totally agree—having users add this manually would be bad DX. We can handle it similarly to how it's done in the pages router.

I could probably add the Partytown snippet in the app router if worker support is enabled, but I still need to figure out how to handle the custom Partytown config since it has to be included before the snippet. At first, I wanted to use `ReactDOM.preload()` for that, but the config is an inline script, so it wouldn’t work. I’m thinking we could use a pattern similar to how scripts are handled, like:

```jsx
<script
  nonce={nonce}
  dangerouslySetInnerHTML={{
    __html: `(self.__next_s=self.__next_s||[]).unshift(${JSON.stringify([
      0,
      { ...restProps, id },
    ])})`,
  }}
/>
```

Just unshifting to the front of the array instead of pushing. Should work!

Reply:
That sounds good to me